### ISSUE TITLE
Removing `self` causes confusion between global method and instance method

### ISSUE THREAD

**Issue by kevinrenskers** [nicklockwood/SwiftFormat#988]
I have this method:

``` swift
public extension Array where Element: Comparable {
    /// Returns the index of the minimum element in the sequence.
    ///
    /// - Returns: the minimum element or nil if it doesn't exist.
    func indexOfMinElement() -> Int? {
        guard let minElement = self.min() else {
            return nil
        }

        return firstIndex(of: minElement)
    }
}
```

SwiftFormat turns `self.min()` into `min()` but then I get this from the compiler: 

```
Use of 'min' treated as a reference to instance method in protocol 'Sequence'
Use 'self.' to silence this warning
Use 'Swift.' to reference the global function
```

**Comment by nicklockwood** [comment]
@kevinrenskers thanks for reporting this. I should probably special-case this, but in the meantime I think you can fix it by adding `--selfrequired min,max` to the swiftformat config.

**Comment by drekka** [comment]
Because this is a specific case that I assume would not appear across a codebase, would it be a better idea to use the disable comments to disable for this line or the whole class?

so `// swiftformat:disable:next selfrequired` if there are not too many spots.

**Comment by nicklockwood** [comment]
@drekka yes, if you don't mind making changes to the code itself

**Comment by kevinrenskers** [comment]
Weirdly enough adding `--selfrequired min` doesn't do anything, SwiftFormat still removes the `self.` and leading to the compiler warning. Same goes for the `// swiftformat:disable:next selfrequired` comment. Just try it out with the exact example code I added to the issue.

**Comment by nicklockwood** [comment]
@kevinrenskers OK, that sounds like another bug 😅

**Comment by drekka** [comment]
Sorry @kevinrenskers I gave you the wrong name. It should have been `// swiftformat:disable:next redundantSelf`

:-)

**Comment by nicklockwood** [comment]
@kevinrenskers fixed in 0.48.12. Using `--selfrequired` should also now work for similar cases that aren't supported out of the box.
